### DISCUSSION TITLE
Issue with ps_events after each module updates logs out the account in the ps_accounts module

### DISCUSSION THREAD

**Discussion by Maofree** [PrestaShop/PrestaShop#33786] [Q&A] [answered]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hi
yesterday I've updated, on 25 websites, ps_events from 2.3.1 to 2.3.2
today the a my customer alerts me telling me that paypal was no longer in the payment list
in fact ps_account had disassociated the addons account

this problem happened on 12 sites out of the 25 to which I made this update.
this thing happened before with another update but in that case I thought it was due to an update of ps_accounts since that's what disconnects and creates problems.
now I understand that the problem is created by each update of the ps_events module

it happened on 12 sites out of the 25 to which I made this update.
this thing happened before with another update but in that case I thought it was due to an update of ps_accounts since that's what disconnects and creates problems.
now I understand that the problem is created by each update of the ps_events module

the problema happens the day after the ps_events upgrade. The same happens some weeks ago

This problem happens on 1.7.8.9 and on 8.1.1 with php 7.4 and 8.1
PrestaShop Account v6.2.6 and with PrestaShop Account v5.4.8

![1](https://github.com/PrestaShop/PrestaShop/assets/653230/02becfb3-b610-48ef-96ad-841cca82c838)
![2](https://github.com/PrestaShop/PrestaShop/assets/653230/b0ddf25a-6429-41f3-bb64-94a15da7199a)

these are similar topic already created, I didn't quite understand this problem
https://github.com/PrestaShop/PrestaShop/issues/33331
https://github.com/PrestaShop/PrestaShop/issues/32237

bye

### Steps to reproduce

1. already installed ps_events 2.3.1
2. already installed ps_accounts
3. wait  24 hours and che if ps_accounts is without associations
4. the problem happens in random mode, not happens each times. On 25 websites only 12 have this problem

### PrestaShop version(s) where the bug happened

8.1.1 and 1.7.8.9

### PHP version(s) where the bug happened

7.4 and 8.1

### If your bug is related to a module, specify its name and its version

ps_events and ps_accounts

### Your company or customer's name goes here (if applicable).

maofree

**Comment by kpodemski** [comment]
Hello @maofree 

As you know, these modules are not part of the open-source project. I let the team responsible for these modules know about your issues.

**Comment by Maofree** [comment]
ok thanks

this error is causing me several problems, because it happens randomly and on different sites that I manage, so every time I'm forced to double check if they are connected and if they weren't to reconnect them. Then paypal stops working and therefore becomes a problem for shopkeepers who don't know about the problem unless notified by customers.
now I know that the day after I update these modules I will have to check them again

**Comment by Maofree** [comment]
Hi
today another company alert me that the paypal is no more present in the payment list
the ps_account is without association

now I'm looking other website and others have the same problem. this had never happened, that they disassociated themselves after a few days that I associated them

they need to find a solution to this problem

![1](https://github.com/PrestaShop/PrestaShop/assets/653230/0b3c0dfa-509b-4e0e-9552-6fe5511a3656)

**Comment by Maofree** [comment]
![1](https://github.com/PrestaShop/PrestaShop/assets/653230/ce9013db-cb0a-435f-afaf-17bc0a46e593)
![4](https://github.com/PrestaShop/PrestaShop/assets/653230/938a9939-6d11-412a-a1d1-a00224f1e541)

at this time was fantastic, 22 websites are disconneted from ps_account and I try to reconneted all of them and I 'm banned from prestashop's server

**Comment by Maofree** [comment]
how do they expect to solve these problems, that I call each company and make them do it for them?
i thought i can't handle 30 sites alone from my pc?
this problem must be solved by force

**Comment by Maofree** [comment]
can I get also the contacts of who runs this crap system?

thanks

**Comment by Maofree** [comment]
now I'm having the companies do the association but the system tells him that there's an error and doesn't let him do it, so I'm forced to use the other form for paypal

**Comment by Maofree** [comment]
are they aware that the system has problems? now I'm having the companies do the association and they can't do it because the system says there is an error.
I can think that I am the problem, because I make associations from my pc on 30 sites and that there is a cookie problem by logging in and out of different accounts in a few minutes. But I don't think that's why

**Comment by gennaris** [comment]
Yes, this is frustrating - I am encountering the same issue.

I got several stores  disconnected out of nowhere last days with the effect that the end customers did not see PayPal/card payments options anymore and the merchant consequentely lost money.

Reconnecting some shops is not always doable as it ends with an error,  some other required me redoing the whole PS Account wizard with huge waste of time.

I have the feeling the problem stems from an external API failing to check the website connection and / or having multiple sites linked to different PS accounts (agencies / merchants) and accessing them from the same pc (that is my case).

Official support just said to get the last version of the PsCorp modules suite, but in my case they were already at the latest version everywhere.

**Comment by Maofree** [comment]
I do it in the same way
but now I do it from each shopkeeper and they get the same error to do the connection from their pc

**Comment by Guillaumelpg** [comment]
Hello,

I am the referent of the ps_accounts module. We have identified an infrastructure issue that may be causing the problems reported here.

We will deploy a patch this afternoon. 

I'll let you know when it's done.

Guillaume

**Comment by Guillaumelpg** [comment, marked as answer]
Hello, 

The patch was deployed yesterday. 

We have ensured that stores that have been dissociated are automatically re-associated.

If this is not the case, we invite you to re-associate your stores.

However, if you encounter an error again, please contact PrestaShop support via this contact form: https://help-center.prestashop.com/en/contact.

Guillaume

**Reply by cdigruttola** [reply]
This is not an answer. It is just a response to write something to keep quite the people. It is not possible that with the centralized account you are not able to stay always online with the server (that are on Firebase if you see the code).

**Comment by Maofree** [comment]
hi
today again this error
all websites again with the same error
now if I enter go inside the module ps_accounts I get this error

![1](https://github.com/PrestaShop/PrestaShop/assets/653230/5c63e21f-a31e-46d5-9152-77e1c5c7975f)

this problem must be fixed
you can't make a module that has various modules connected to it, tens of thousands of sites and if it has a problem everything goes haywire.
there are times when you have problems and everything disconnects

this happen on last version of these modules and last prestashop's version

**Comment by Maofree** [comment]
now it return to be conneted, but now you can see what is causing this problem to haywire, it is a curl connection

**Comment by Maofree** [comment]
automatic reconnection is not working, because I have 20 sites that are no longer connected with that ps_accounts module

**Comment by Maofree** [comment]
![1](https://github.com/PrestaShop/PrestaShop/assets/653230/fb53fe4f-15f7-4a51-ac63-23c697185e2b)

now again this error in ps_accounts

**Comment by Maofree** [comment]
@Guillaumelpg 
can you contact me? because it is necessary to fix this problem and I can help you to give informations or access to my websites

**Reply by Guillaumelpg** [reply]
Hello @maofree 
Yes I am available on the following email address : guillaume.lepage@prestashop.com 
If you can give me as many details as possible, I can give them to the tech team so that we can understand why this dissociation error persists.
I've already passed on the info shared on this thread, we are currently analyzing

**Reply by Maofree** [reply]
Hi
ok the next issue I'll report more info to you
however that curl error should help you to fix this problem
the ps_accounts or ps_events module shouldn't be so crucial for the functioning of the modules connected to it
- unfortunately there will always be errors in the servers and this should not create critical errors in the connected modules and above all not cause disconnection. Also many other modules such as those for Amazon, Ebay, paypal, stripe connect remotely via API to external servers and if they have problems they do not disconnect, if they too had these problems, it would be a delirium to have to check them all every week to see if they are functioning and connected

**Reply by cdigruttola** [reply]
Just to add that the info in this thread are the most precise. In the attached file you can see more info
[ps_accounts-2023-09-06.log](https://github.com/PrestaShop/PrestaShop/files/12546147/ps_accounts-2023-09-06.log)

**Comment by Maofree** [comment]
today I cannot finish the verification to do a new associations like in the last weeks. It is blocked to this step

![1](https://github.com/PrestaShop/PrestaShop/assets/653230/c7d1a03b-5bda-4308-b346-fa225a5e5c2f)

![1](https://github.com/PrestaShop/PrestaShop/assets/653230/1bb032ff-7932-4801-a5c6-d0d259fe0429)

**Comment by Maofree** [comment]
![1](https://github.com/PrestaShop/PrestaShop/assets/653230/46ddd1ed-263e-4547-9222-896562f6fc4e)

nothing is working today and the sites are losing orders

**Comment by gennaris** [comment]
I confirm there are still issues for me aswell. 

Just a little while ago, 5 websites I manage disconnected from ps_checkout, and reconnection is not working.

There is a problem with the ps_accounts API ( $accountsService->isAccountLinked() returns false), and nothing is working.

The choice to base the operation of a payment module on a recurring check of external APIs doesn't seem like a brilliant idea to me...

**Comment by Maofree** [comment]
thanks @gennaris  so I understand that the problem depends on the Prestashop servers and it happens at the same time on all sites, so the double shop has nothing to do with it or if the association is made from a single PC or from those of the shopkeepers. It always happens at the same time on many sites, so the cause is the curl connections that do not receive a response from the prestashop servers, perhaps because they are not able to handle all these calls

prestashop can't centralize everything on a single module, adding other problems to modules that connect remotely to PayPal, Google, Facebook servers... and if their servers then go haywire, everything on tens of thousands of websites crashes.

**Comment by Maofree** [comment]
I've just finished relinking them in the ps_accounts module
for 3 hours and more that module stopped working. Only in the last few months has it started giving problems
I hope that the errors reported can solve this problem altogether
you certainly cannot continue to manage systems as important as paypal in this way
then the automatic reconnection doesn't work
honestly it doesn't make sense to disconnect a module just because there are problems connecting to a server.
the module should remember the connection and the servers should do the same, a connection error must be taken into account and not become critical as now

**Comment by Maofree** [comment]
HI
the problem is not solved so this discussion should stay open

**Comment by cdigruttola** [comment]
Please reopen the discussion, the problem is not solved

**Comment by ILDaviz** [comment]
Unresolved problem randomly logs me out to ps_checkout and ps_account

**Comment by adviva** [comment]
This is seriously frustrating. An official payment module like this really cannot have this kind of problem.
Everything is up to date. No Cloudflare. Still impossible to re-link the shop.
![Screenshot 2023-09-07 at 15 48 32](https://github.com/PrestaShop/PrestaShop/assets/26090185/4a71c2a2-1057-46bd-b7ee-02c1b007d491)
![Screenshot 2023-09-07 at 15 46 41](https://github.com/PrestaShop/PrestaShop/assets/26090185/b49cc5c3-d080-4dbd-88c3-4877f9f874a1)

**Comment by Maofree** [comment]
Hi @Guillaumelpg 
today I've done an upgrade from prestashop 1.7.8.9 to 8.1.1
the upgrade with 1-click got an error about a a file not present in the module ps_accounts
it is the second time that I get this error, in both cases it is necessary to reinstall ps_accounts (or remove and install it after the upgrade) before to do a new test to upgrade to the 8.1.1

at this moment also it is not present ps_accounts on Addons

![1111](https://github.com/PrestaShop/PrestaShop/assets/653230/dafd47db-e957-4351-a76a-7e0a838cf133)


bye

**Reply by Guillaumelpg** [reply]
Hello @maofree 

Thanks for your feedback. I will share the information with the team. 

Concerning the PS Account module which is not visible on Addons, we will make it available in the coming days/weeks.

**Reply by Guillaumelpg** [reply]
@maofree I've received some news from the team. 

Using 1-click, there is no update of the modules, which explains the error because versions 1.7.X of the stores are compatible with version 5.X of ps_accounts and versions 8.X are compatible with version 6.X of the module

**Reply by Maofree** [reply]
so before the upgrade it is better to remove it
I've seen that could be ok, reinstall it, probably that file is missing from past versions

**Comment by Maofree** [comment]
Hi @Guillaumelpg 
today I get a new problem with this module
![1](https://github.com/PrestaShop/PrestaShop/assets/653230/5a33a442-10fb-49e9-a5cc-24c9781c4cd7)

I sent you an email with the admin url to see this problem
it is the first time that I get this problem

and it's the first time I let the customer do the configuration, so I don't know what he did differently, but usually they ask for company data, so it seems strange to me that this problem comes up.
Now how do I remove this page?
